pkg/shp/util/logs.go

```diff
@@ -0,0 +1,32 @@
+package util
+
+import (
+	"bytes"
+	"context"
+	"io"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+// GetPodLogs returns log output of the k8s container provided by pod and name
+func GetPodLogs(ctx context.Context, client *kubernetes.Clientset, pod corev1.Pod, container string) (string, error) {
+	podLogOpts := corev1.PodLogOptions{
+		Container: container,
+	}
+	req := client.CoreV1().Pods(pod.Namespace).GetLogs(pod.Name, &podLogOpts)
+	podLogs, err := req.Stream(ctx)
+	if err != nil {
+		return "", err
+	}
+	defer podLogs.Close()
+
+	buf := new(bytes.Buffer)
+	_, err = io.Copy(buf, podLogs)
```

Comment:
actually we use io.Copy in the `oc start-build ... -F` analogous flavor of this @coreydaley  .... I think I would be more inclined with "copying" :-) from pre-existing impls

on that point  @alicerum since you already have the individual `run` and `logs` verbs here, I think it would be easy to add a `-F` option to `run` that combines the two.

the magic client calls are
1) your existing use of `GetLogs` .... i.e. https://github.com/kubernetes/client-go/blob/c8af929f7f6f3357249ffeb733e51e47b7459149/kubernetes/typed/core/v1/pod_expansion.go#L34
2) then call `Stream()` from that return ... i.e. https://github.com/kubernetes/client-go/blob/6a42ca30bacfc4ec1b1eb9ce116d875365ee19b4/rest/request.go#L763

The analogous application / use of this via `oc start-build -F` (the build log client basically proxies to the generic k8s pod log client):  https://github.com/openshift/oc/blob/master/pkg/cli/startbuild/startbuild.go#L490-L514

Reply:
@gabemontero thank you! i'll check this out and see if it can be all used easily together

Reply:
@gabemontero @otaviof 
I think that it's still too much of a change for a PR review change. Can we do it in separate PR for the [build-207](https://issues.redhat.com/browse/BUILD-207) issue?